File: surf-friends/src/main/java/dev/slne/surf/friends/command/subcommand/FriendRequestListCommand.java
package dev.slne.surf.friends.command.subcommand;

import dev.jorel.commandapi.CommandAPICommand;
import dev.slne.surf.friends.FriendManager;
import dev.slne.surf.friends.SurfFriendsPlugin;
import it.unimi.dsi.fastutil.objects.ObjectList;
import java.util.UUID;
import net.kyori.adventure.text.minimessage.MiniMessage;
import org.bukkit.Bukkit;

public class FriendRequestListCommand extends CommandAPICommand {
  public FriendRequestListCommand(String commandName) {
    super(commandName);

    executesPlayer((player, args) -> {
      String message = "Du hast keine Freunde.";

      ObjectList<UUID> requests = FriendManager.instance().getFriendRequests(player.getUniqueId());
      StringBuilder builder = new StringBuilder("Du hast aktuell <yellow>" + requests.size() + "<white> Freundschaftsanfragen offen: ");
      int current = 0;

      for (UUID uuid : requests) {
        current ++;

        if(current == requests.size()) {
          builder.append("<white>").append(Bukkit.getOfflinePlayer(uuid).getName());
        }else{
          builder.append("<white>").append(Bukkit.getOfflinePlayer(uuid).getName()).append("<gray>, ");
        }
      }

      if(!requests.isEmpty()) {
        message = builder.toString();
      }

      player.sendMessage(SurfFriendsPlugin.getPrefix().append(MiniMessage.miniMessage().deserialize(message)));
    });
  }
}

File: surf-friends/src/main/java/dev/slne/surf/friends/command/subcommand/FriendRequestListCommand.kt
package dev.slne.surf.friends.command.subcommand

import com.github.shynixn.mccoroutine.bukkit.launch
import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.executors.CommandArguments
import dev.jorel.commandapi.executors.PlayerCommandExecutor
import dev.slne.surf.friends.FriendManager
import dev.slne.surf.friends.plugin
import dev.slne.surf.friends.prefix
import net.kyori.adventure.text.minimessage.MiniMessage
import org.bukkit.Bukkit
import org.bukkit.entity.Player

class FriendRequestListCommand(commandName: String) : CommandAPICommand(commandName) {

    init {
        executesPlayer(PlayerCommandExecutor { player: Player, _: CommandArguments? ->
            plugin.launch {
                val requests = FriendManager.getFriendRequests(player.uniqueId)

                val message = buildString {
                    append("Du hast aktuell <yellow>").append(requests.size)
                        .append("<white> Freundschaftsanfragen offen: ")

                    requests.forEachIndexed { index, uuid ->
                        if (index == requests.size - 1) {
                            append("<white>").append(Bukkit.getOfflinePlayer(uuid).name)
                        } else {
                            append("<white>").append(Bukkit.getOfflinePlayer(uuid).name)
                                .append("<gray>, ")
                        }
                    }
                }

                player.sendMessage(
                    prefix.append(MiniMessage.miniMessage().deserialize(message))
                )
            }
        })
    }
}
